src/memray/commands/attach.py

```diff
@@ -332,11 +340,19 @@ def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None
             live_port = _get_free_port()
             destination = memray.SocketDestination(server_port=live_port)
 
+        if args.aggregate and not hasattr(args, "output"):
+            parser.error("Can't use aggregated mode without an output file.")
+
+        file_format = (
+            f"file_format={FileFormat.AGGREGATED_ALLOCATIONS}" if args.aggregate else ""
+        )
```

Comment:
Oooh, this is interesting. This winds up passing `file_format=1`, because:
```py
>>> FileFormat.AGGREGATED_ALLOCATIONS
<FileFormat.AGGREGATED_ALLOCATIONS: 1>
>>> f"file_format={FileFormat.AGGREGATED_ALLOCATIONS}"
'file_format=1'
```

I'm surprised that works. I think it would be more robust to do:
```py
        file_format = (
            "file_format=memray.FileFormat.AGGREGATED_ALLOCATIONS" if args.aggregate else ""
        )
```

I'd rather pass around the enum by name than by value, in case a future version of the API changes the value associated with the enum, or changes the `Tracker` constructor to reject ints and only accept enum members.

I'd like to see an integration test that exercises this, too - that should be as easy as [parametrizing](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) our [current `attach` test](https://github.com/bloomberg/memray/blob/main/tests/integration/test_attach.py#L55) with a new flag for whether or not to use aggregated mode, passing `--aggregate` in the `attach_cmd` when the flag is set, and asserting that `get_allocation_records()` fails when `--aggregate` was passed but succeeds when it wasn't, and that `get_high_watermark_allocation_records()` works in both cases.